### PR TITLE
fuzzy search, login, added envivs

### PR DESCRIPTION
**Pull request by tairezzzz**
fuzzy search, login, added envivs

### CODE REVIEW

File: src/app/index.config.js [hunk 1]
```diff
@@ -16,4 +16,11 @@ export function config ($logProvider, toastrConfig, $mdThemingProvider, cfpLoadi
 
   // Load Bar
   cfpLoadingBarProvider.includeSpinner = false;
+
+  // Google+
+  GooglePlusProvider.init({
+    clientId: '867651060370-4oqj1vdb06deotdvek82riivej3q4pmk.apps.googleusercontent.com'
```
Comment (MicahZoltu) [inline src/app/index.config.js]: Can you set the `scopes` to `profile` here?  Details can be found at https://github.com/mrzmyr/angular-google-plus/issues/6 with a specific example at https://github.com/mrzmyr/angular-google-plus/issues/6#issuecomment-52639212.

Right now it is requesting the `https://www.googleapis.com/auth/plus.login` scope which is over-asking.

**Comment by MicahZoltu** [comment]
The redirect doesn't seem to be working locally.  Is there a trick to getting it setup?  After I authorize in the pop-up it doesn't go away.  I suspect that the problem is the redirect isn't setup correctly for local development, but I don't see where it is setup in the PR.

**Comment by MicahZoltu** [comment]
In Chrome the pop-up is blocked.  I believe this happens when the pop-up isn't triggered directly off of a click.  For example, if you trigger the pop-up as part of a timer execution or a promise callback chrome will block it.

I suspect that the problem is at https://github.com/Zoltu/bags-admin-client/pull/8/files#diff-d987f9cee335a281b239ad6baa3c1657R12.  I don't know Angular, but I would assume there is some way to have the pop-up called as part of the direct call tree of the user interaction (click)?

**Comment by MicahZoltu** [comment]
Waiting to merge this until I can verify locally that the login is working.  Right now the login isn't working locally, and also I can't navigate to any of the internal routes (I suspect because login is failing).

The other two comments can wait until a future PR, or they can be addressed as part of this PR.

File: src/app/index.config.js (hunk 1 above)
Comment (tairezzzz) [inline src/app/index.config.js]: Done

**Comment by tairezzzz** [comment]
There is no trick to manage redirect locally, maybe you have to make to make sure that you've setup right google application to work with your localhost. I just don't send redirect_uri, because it's not required. I saw that it appears with value 'postmessage' after calling gapi.auth.authorize(params, callback). But I don't think that's reason for your error. Maybe you can send console screen with error messages?

**Comment by tairezzzz** [comment]
If you don't want this popup to be blocked we just should add some element like login button somewhere. It allows to avoid popup, but will require additional action from user. More info here:  https://developers.google.com/api-client-library/javascript/reference/referencedocs#methods. We can always allow popups as on that screen.

**Comment by MicahZoltu** [comment]
For the popup, I would rather a login button than a blocked pop-up.  Can you make it so the pop-up appears as soon as the user attempts to interact with any of the nav items on the left-bar?  This would prevent needing a special login button but also should resolve the pop-up blocker issue.

The problem with the pop-up blocker is that my admins may not notice it (it is very subtle in chrome) and it can quickly become frustrating for them when the UI just doesn't work.

**Comment by MicahZoltu** [comment]
> Maybe you can send console screen with error messages?

I am not seeing any error in the console output for either the pop-up or the main page.  After logging in to Google, the pop-up just stays popped up with a completely blank page.  I am never redirected back to the application.

After some more troubleshooting, the problem appears to be related to the pop-up blocker.  When I disabled pop-up blocking for the application, everything works.  When pop-up blocking is enabled, the pop-up doesn't appear to function correctly even if I click on the pop-up blocker dialog and click the link to manually open the pop-up.

So I guess doing something to dodge the pop-up blocker will resolve both the issue of it being difficult for my admins and also not working correctly.  :)

**Comment by MicahZoltu** [comment]
In the latest version, when I try to create a new product I get this error instead of the creation dialog:
image

I am still able to edit products though.

**Comment by MicahZoltu** [comment]
When clicking edit button on a product there appears to be a `console.log` left in.  That should be cleaned up (not sure if it is related to this PR or not, just noticed it when debugging the broken Create button).

Similarly, when the edit pop-up is closed, `undefined` is logged.

**Pull request merged**